src/prog_models/data_models/dmd.py

```diff
@@ -420,7 +420,7 @@ def next_state(self, x, u, _):
         if self.B.shape[1] != 0:
            x.matrix += np.matmul(self.B, u.matrix)
         
-        return x   
+        return self.StateContainer(x.matrix)
```

Comment:
Why is this needed? If x has matrix, isn't it already guaranteed to be a state container?

Reply:
fixed as we discussed

Reply:
Didn't we decide to just add a comment here instead of creating a new state container?